osu.Game/Rulesets/UI/Playfield.cs

```diff
@@ -47,6 +47,11 @@ public abstract class Playfield : CompositeDrawable
         /// </summary>
         public readonly BindableBool DisplayJudgements = new BindableBool(true);
 
+        /// <summary>
+        /// Whether this <see cref="Playfield"/> has failed.
+        /// </summary>
+        protected readonly BindableBool HasFailed = new BindableBool();
```

Comment:
why does playfield have state?

Reply:
so in `OsuPlayfield`, it binds `HasFailed` of `FollowPointRenderer` to this, so the follow points fail

Reply:
next question: why does player not bind to this rather than calling `Fail()`? currently fail is potentially not safe against multiple calls.

Reply:
Good idea

Reply:
Should that be applied to DrawableHitObject as well?